### PR TITLE
Add Configuration for Highlight Related Feature

### LINKED ISSUE #9618: Add config settings for highlight_related
https://github.com/rust-analyzer/rust-analyzer/blob/a2f83c956e277b23e20ea2f4f0ac8c5034fe2507/crates/ide/src/highlight_related.rs#L48-L52
https://github.com/rust-analyzer/rust-analyzer/blob/a2f83c956e277b23e20ea2f4f0ac8c5034fe2507/crates/ide/src/highlight_related.rs#L23-L27
Just a bool settings:
`highlight_related.references`
`highlight_related.exit_points`
`highlight_related.break_points`
`highlight_related.yield_points`

### PR DESCRIPTION
**Pull request by kdelorey**
# Summary
Adds basic configuration that allows you to control when the highlight related feature is activated. You can control this for references, break points, exit points, and yield points.

Resolves #9618 

config

### CODE REVIEW

File: editors/code/src/config.ts [hunk 1]
```diff
@@ -172,6 +172,15 @@ export class Config {
...
     }
 
+    get highlightRelated() {
+        return {
+            references: this.get<boolean>("highlightRelated.references"),
+            exitPoints: this.get<boolean>("highlightRelated.exit_points"),
+            breakPoints: this.get<boolean>("highlightRelated.exit_points"),
+            yieldPoints: this.get<boolean>("highlightRelated.yield_points")
+        };
+    }
```
Comment (matklad) [inline editors/code/src/config.ts:182]: I think this is dead code and should be removed?

Looks perfect to me otherwise, thanks!

bors d+

Comment (kdelorey) [reply editors/code/src/config.ts:182]: Makes sense to me to remove, I was just mimicking other configuration here.

Comment (matklad) [reply editors/code/src/config.ts:182]: other stuff is used on the VS Code side as well I belive. Although if someth things are unsued, might makes sense to remove thouse

**Pull request merged**